src/sentry/api/endpoints/organization_integration_details.py

```diff
@@ -4,27 +4,42 @@
     OrganizationEndpoint, OrganizationIntegrationsPermission
 )
 from sentry.api.serializers import serialize
-from sentry.models import Integration, OrganizationIntegration
+from sentry.models import OrganizationIntegration, ProjectIntegration
 
 
 class OrganizationIntegrationDetailsEndpoint(OrganizationEndpoint):
     permission_classes = (OrganizationIntegrationsPermission, )
 
     def get(self, request, organization, integration_id):
-        integration = Integration.objects.get(
-            organizations=organization,
-            id=integration_id,
+        integration = OrganizationIntegration.objects.get(
```

Comment:
definitely was a problem before, but maybe you could wrap this in a try, except `OrganizationIntegration.DoesNotExist` while you're in here

Reply:
9a4753f